.bumpversion.toml

```diff
@@ -1,57 +1,86 @@
 current_version = "0.10.0-beta.1"
+ignore_missing_files = false
+ignore_missing_version = false
+message = "Bump version: {current_version} → {new_version}"
 parse = """(?x)
     (?P<major>0|[1-9]\\d*)\\.
     (?P<minor>0|[1-9]\\d*)\\.
     (?P<patch>0|[1-9]\\d*)
     (?:-(?P<pre_l>[a-zA-Z-]+)\\.(?P<pre_n>0|[1-9]\\d*))?
 """
+regex = false
+replace = "{new_version}"
+search = "{current_version}"
 serialize = [
-    "{major}.{minor}.{patch}-{pre_l}.{pre_n}",
-    "{major}.{minor}.{patch}",
+  "{major}.{minor}.{patch}-{pre_l}.{pre_n}",
+  "{major}.{minor}.{patch}",
 ]
-search = "{current_version}"
-replace = "{new_version}"
-regex = false
-ignore_missing_version = false
-ignore_missing_files = false
-tag = true
 sign_tags = false
-tag_name = "v{new_version}"
+tag = true
 tag_message = "Bump version: {current_version} → {new_version}"
-allow_dirty = true
-commit = true
-message = "Bump version: {current_version} → {new_version}"
-commit_args = ""
+tag_name = "v{new_version}"
 
 [tool.bumpversion.parts.pre_l]
-values = ["beta", "final"]
 optional_value = "final"
+values = ["beta", "final"]
 
 [[tool.bumpversion.files]]
 filename = "node/package.json"
-search = "\"version\": \"{current_version}\","
 replace = "\"version\": \"{new_version}\","
+search = "\"version\": \"{current_version}\","
 
 [[tool.bumpversion.files]]
 filename = "nodejs/package.json"
-search = "\"version\": \"{current_version}\","
 replace = "\"version\": \"{new_version}\","
+search = "\"version\": \"{current_version}\","
 
 # nodejs binary packages
 [[tool.bumpversion.files]]
 glob = "nodejs/npm/*/package.json"
-search = "\"version\": \"{current_version}\","
 replace = "\"version\": \"{new_version}\","
+search = "\"version\": \"{current_version}\","
 
 # Cargo files
 # ------------
 [[tool.bumpversion.files]]
 filename = "rust/ffi/node/Cargo.toml"
-search = "\nversion = \"{current_version}\""
 replace = "\nversion = \"{new_version}\""
+search = "\nversion = \"{current_version}\""
 
 [[tool.bumpversion.files]]
 filename = "rust/lancedb/Cargo.toml"
-search = "\nversion = \"{current_version}\""
 replace = "\nversion = \"{new_version}\""
+search = "\nversion = \"{current_version}\""
+
+# Java files
+# Consider using mvn version management if java project becomes more complex
+# ------------
+[[tool.bumpversion.files]]
+glob = "java/pom.xml"
+replace = """
+    <groupId>com.lancedb</groupId>
+    <artifactId>lancedb-parent</artifactId>
+    <version>{new_version}</version>
+"""
+search = """
+    <groupId>com.lancedb</groupId>
+    <artifactId>lancedb-parent</artifactId>
+    <version>{current_version}</version>
+"""
+
+[[tool.bumpversion.files]]
+glob = "java/*/**/pom.xml"
+replace = """
+        <groupId>com.lancedb</groupId>
+        <artifactId>lancedb-parent</artifactId>
+        <version>{new_version}</version>
+"""
+search = """
+        <groupId>com.lancedb</groupId>
+        <artifactId>lancedb-parent</artifactId>
+        <version>{current_version}</version>
+"""
```

Comment:
This approach may be broken easily since it does not take care of the complicated java dependencies

https://github.com/LuQQiu/lancedb/blob/f08af7e747ade81af988b594057402336083902c/ci/bump_version.sh#L13

Another approach is to leverage "mvn versions" command, but it does not integrate seamlessly with python bump-my-version.